lang/step.md

```diff
@@ -395,9 +406,12 @@ impl<M: Memory> Machine<M> {
         }
 
         // Advance the PC for this stack frame.
-        self.mutate_cur_frame(|frame| {
-            frame.jump_to_block(next_block);
-        });
+        if fn_might_return {
+            let next_block = next_block.unwrap();
+            self.mutate_cur_frame(|frame| {
+                frame.jump_to_block(next_block);
+            });
+        }
```

Comment:
Okay here we have a problem. If `next_block` is `None` we cannot jump to the next block already. And we don't want UB *now* in that case, we want UB only if the callee ever returns.

In Miri this works because we only jump to the next frame when the callee returns. Maybe we should do that here, too?